File: frontend/src/services/NotificationService.js
import api from "../api/axios";

const NotificationService = {

    getNotifications() {

        return api.get("/notifications");

    },

    getUnreadCount() {

        return api.get("/notifications/unread-count");

    },

    markAsRead(id) {

        return api.put(`/notifications/${id}/read`);

    },

    markAllRead() {

        return api.put("/notifications/read-all");

    },

};

export default NotificationService;

File: backend/src/main/java/com/soubhagya/pingme/service/NotificationService.java
package com.soubhagya.pingme.service;

import com.soubhagya.pingme.dto.response.NotificationResponse;

import java.util.List;

public interface NotificationService {

    List<NotificationResponse> getNotifications(String email);

    long getUnreadCount(String email);

    void markAsRead(Long notificationId, String email);

    void markAllRead(String email);

    void createFriendRequestNotification(
            Long recipientId,
            Long senderId,
            String senderName
    );

    void createFriendRequestAcceptedNotification(
            Long recipientId,
            Long accepterId,
            String accepterName
    );

    void createMessageNotification(
            Long recipientId,
            Long senderId,
            String senderName,
            String preview
    );

    void createMissedCallNotification(
            Long recipientId,
            Long callerId,
            String callerName,
            String callType
    );

}